test/test-eta.js

```diff
@@ -0,0 +1,843 @@
+      t.strictEqual(eta.render(fs.readFileSync('./templates/index.eta', 'utf8'), data), body.toString())
+      fastify.close()
+    })
+  })
+})
+
+test('reply.view with eta engine and html-minifier', t => {
+  t.plan(6)
+  const fastify = Fastify()
+  const eta = require('eta')
+  const data = { text: 'text' }
+
+  fastify.register(require('../index'), {
+    engine: {
+      eta: eta
+    },
+    options: {
+      useHtmlMinifier: minifier,
+      htmlMinifierOptions: minifierOpts
+    }
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.view('templates/index.eta', data)
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    sget({
+      method: 'GET',
+      url: 'http://localhost:' + fastify.server.address().port
+    }, (err, response, body) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 200)
+      t.strictEqual(response.headers['content-length'], '' + body.length)
+      t.strictEqual(response.headers['content-type'], 'text/html; charset=utf-8')
+      t.strictEqual(minifier.minify(eta.render(fs.readFileSync('./templates/index.eta', 'utf8'), data), minifierOpts), body.toString())
+      fastify.close()
+    })
+  })
+})
+
+test('reply.view with eta engine and includeViewExtension property as true', t => {
+  t.plan(6)
+  const fastify = Fastify()
+  const eta = require('eta')
+  const data = { text: 'text' }
+
+  fastify.register(require('../index'), {
+    engine: {
+      eta: eta
+    },
+    includeViewExtension: true
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.view('templates/index', data)
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    sget({
+      method: 'GET',
+      url: 'http://localhost:' + fastify.server.address().port
+    }, (err, response, body) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 200)
+      t.strictEqual(response.headers['content-length'], '' + body.length)
+      t.strictEqual(response.headers['content-type'], 'text/html; charset=utf-8')
+      t.strictEqual(eta.render(fs.readFileSync('./templates/index.eta', 'utf8'), data), body.toString())
+      fastify.close()
+    })
+  })
+})
+
+test('reply.view with eta engine, template folder specified, include files (eta and html) used in template, includeViewExtension property as true', t => {
+  t.plan(7)
+  const fastify = Fastify()
+  const eta = require('eta')
+  const templatesFolder = path.join(__dirname, '../templates')
+  const options = {
+    views: templatesFolder // must be put to make tests (with include files) working ...
+  }
+  const data = { text: 'text' }
+
+  fastify.register(require('../index'), {
+    engine: {
+      eta: eta
+    },
+    includeViewExtension: true,
+    templates: templatesFolder,
+    options: options
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.type('text/html; charset=utf-8').view('index-linking-other-pages', data) // sample for specifying with type
+    // reply.view('index-with-includes', data)
```

Comment:
why is this commented?

Reply:
TBH, I'm not really sure (these tests are just modified from EJS'). My guess is that it was an earlier version of the test -- I'll remove it.